iib/web/models.py

```diff
@@ -365,8 +409,52 @@ def from_json(cls, kwargs):
         if current_user.is_authenticated:
             request_kwargs['user'] = current_user
 
+    @classmethod
+    def from_add_json(cls, kwargs):
+        """ Handles JSON requests for Add API endpoint """
+        request_kwargs = deepcopy(kwargs)
+
+        bundles = request_kwargs.pop('bundles', [])
+        if (
+            not isinstance(bundles, list)
+            or len(bundles) == 0
+            or any(not item or not isinstance(item, str) for item in bundles)
+        ):
+            raise ValidationError(f'"bundles" should be a non-empty array of strings')
+
+        request_kwargs['bundles'] = [
+            Image.get_or_create(pull_specification=item) for item in bundles
+        ]
+
+        cls._from_json(
+            request_kwargs,
+            additional_required_params=['bundles'],
+            additional_optional_params=['from_index'],
+        )
+
         request_kwargs['type'] = RequestTypeMapping.__members__['add'].value
+        request = cls(**request_kwargs)
+        request.add_state('in_progress', 'The request was initiated')
+        return request
+
+    @classmethod
+    def from_remove_json(cls, kwargs):
+        """ Handles JSON requests for Remove API endpoint """
+        request_kwargs = deepcopy(kwargs)
+
+        operators = request_kwargs.pop('operators', [])
+        if (
+            not isinstance(operators, list)
+            or len(operators) == 0
+            or any(not item or not isinstance(item, str) for item in operators)
+        ):
+            raise ValidationError(f'"operators" should be a non-empty array of strings')
+
+        request_kwargs['operators'] = [Operator.get_or_create(name=item) for item in operators]
```

Comment:
If you don't `pop` the `operators` key, you can perform this after the `cls._from_json` call, which means the database isn't touched until all validation has occurred which is nice.